baselines/ppo2/layers.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import tensorflow as tf
+
+from baselines.a2c.utils import ortho_init, fc
+from baselines.common.models import register
+
+
+@register("ppo_lstm")
+def ppo_lstm(nlstm=128, layer_norm=False):
+    def network_fn(input, mask):
+        memory_size = nlstm * 2
+        nbatch = input.shape[0]
+        mask.get_shape().assert_is_compatible_with([nbatch])
+        state = tf.Variable(np.zeros([nbatch, memory_size]),
+                            name='lstm_state',
+                            trainable=False,
+                            dtype=tf.float32,
+                            collections=[tf.GraphKeys.LOCAL_VARIABLES])
+
+        def _network_fn(input, mask, state):
+            input = tf.layers.flatten(input)
+            mask = tf.to_float(mask)
+
+            if layer_norm:
+                h, next_state = lnlstm(input, mask, state, scope='lnlstm', nh=nlstm)
+            else:
+                h, next_state = lstm(input, mask, state, scope='lstm', nh=nlstm)
+            return h, next_state
+
+        return state, _network_fn
+
+    return network_fn
+
+
+@register("ppo_cnn_lstm")
+def ppo_cnn_lstm(nlstm=128, layer_norm=False, pad='VALID', **conv_kwargs):
+    def network_fn(input, mask):
+        memory_size = nlstm * 2
+        nbatch = input.shape[0]
+        mask.get_shape().assert_is_compatible_with([nbatch])
+        state = tf.Variable(np.zeros([nbatch, memory_size]),
+                            name='lstm_state',
+                            trainable=False,
+                            dtype=tf.float32,
+                            collections=[tf.GraphKeys.LOCAL_VARIABLES])
+
+        def _network_fn(input, mask, state):
+            mask = tf.to_float(mask)
+            initializer = ortho_init(np.sqrt(2))
+
+            h = tf.contrib.layers.conv2d(input,
```

Comment:
I think you can use baselines.common.nature_cnn here to create a stack of conv layers

Reply:
Okay 👍

Reply:
replaced at 36aadd6